frontend/src/model/registerContext.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useContext, useState } from 'react'
+
+interface RegisterContextContent {
+  mail: string
+  password: string
+}
+
+interface RegisterContextUpdate {
+  setMailAndPassword: (mail: string, password: string) => void
+}
+
+const RegisterContext = React.createContext<RegisterContextContent>({
+  mail: '',
+  password: '',
+})
+const RegisterUpdateContext = React.createContext<RegisterContextUpdate>({
+  setMailAndPassword: () => null,
+})
+
+export const useRegister = () => {
+  return useContext(RegisterContext)
+}
+
+export const useRegisterUpdate = () => {
+  return useContext(RegisterUpdateContext)
+}
+
+export const RegisterProvider: React.FC = ({ children }) => {
+  const [user, setUser] = useState({ mail: '', password: '' })
+
+  function setMailAndPassword(mail: string, password: string) {
+    setUser({ mail, password })
+  }
+
+  return (
+    <RegisterContext.Provider value={user}>
+      <RegisterUpdateContext.Provider
+        value={{ setMailAndPassword: setMailAndPassword }}
```

Comment:
```suggestion
  return (
    <RegisterContext.Provider value={user}>
      <RegisterUpdateContext.Provider
        value={{ setMailAndPassword: setUser }}
```
why you wrap the function in a function?

Reply:
Because `setUser` is of type `Dispatch<SetStateAction<{ mail: string; password: string; }>>` which I can't assign to my interface. But setting this as the interface property is also not the greatest way, which is why I used a wrapper around it for simpler types.